### PR TITLE
"added a subtopic on uses of stacks"

### PR DESCRIPTION
**Pull request by rgolu**
- [x ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [ x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ x ] My pull request targets the `master` branch of freeCodeCamp.
- [x ] None of my changes are plagiarized from another source without proper attribution.
- [ x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

File: client/src/pages/guide/english/cplusplus/stack/index.md [hunk 1]
```diff
@@ -163,3 +163,14 @@ int main ()
     2
     1
     Out of loop
+##Uses of Stack
```
Comment (ezioda004) [inline client/src/pages/guide/english/cplusplus/stack/index.md]: Please Fix the heading, there should be a space b/w # and letter.

File: client/src/pages/guide/english/cplusplus/stack/index.md [hunk 2]
```diff
@@ -163,3 +163,14 @@ int main ()
...
+
+1. Expression Evaluation and Conversion.
+   stacks are used to evaluate and convert expressions like prefx,postfix and infix expression. 
+2. In Recursive functions to keep information about the active functions or subroutines.
+3. In Backtracking, as in DFS algorithm.
+4. Memory management, run-time environment for nested language features. etc
+
+
+**
```
Comment (ezioda004) [inline client/src/pages/guide/english/cplusplus/stack/index.md]: Remove this extra character.

File: client/src/pages/guide/english/cplusplus/stack/index.md [hunk 3]
```diff
@@ -163,3 +163,14 @@ int main ()
     2
     1
     Out of loop
+##Uses of Stack
+
+1. Expression Evaluation and Conversion.
+   stacks are used to evaluate and convert expressions like prefx,postfix and infix expression.
```
Comment (ezioda004) [inline client/src/pages/guide/english/cplusplus/stack/index.md]: Can you add whitespace properly after `,`?

File: client/src/pages/guide/english/cplusplus/stack/index.md (hunk 1 above)
Comment (rgolu) [reply client/src/pages/guide/english/cplusplus/stack/index.md]: fixed it.

File: client/src/pages/guide/english/cplusplus/stack/index.md (hunk 3 above)
Comment (rgolu) [reply client/src/pages/guide/english/cplusplus/stack/index.md]: ok

File: client/src/pages/guide/english/cplusplus/stack/index.md (hunk 2 above)
Comment (rgolu) [reply client/src/pages/guide/english/cplusplus/stack/index.md]: ok

**Pull request merged**